### PR TITLE
feat: add sticky filter header for insight tab

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-2905]

#### iOS

https://user-images.githubusercontent.com/3513494/124196970-30b5a800-dad6-11eb-9458-40f49c1e575a.mp4

#### Android

https://user-images.githubusercontent.com/3513494/124197199-afaae080-dad6-11eb-9246-3e31aa0c13ea.mp4

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

[FX-2905]: https://artsyproduct.atlassian.net/browse/FX-2905

### CODE REVIEW

File: src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx [hunk 1]
```diff
@@ -57,40 +58,31 @@ export const ArtistInsights: React.FC<ArtistInsightsProps> = (props) => {
...
+            key: "market-stats",
+            content: <MarketStatsQueryRenderer artistInternalID={artist.internalID} environment={relay.environment} />,
+          },
+          { key: "filter-header", content: <ArtistInsightsFilterHeader onFilterPress={openFilterModal} /> },
+          {
+            key: "content",
+            content: <ArtistInsightsAuctionResultsPaginationContainer artist={artist} scrollToTop={scrollToTop} />,
+          },
+        ]}
+      />
```
Comment (dimatretyak) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx:85]: @MounirDhahri @brainbicycle @olerichter00 
Is this a good way to implement a sticky header? What do you think?

File: src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx [hunk 2]
```diff
@@ -29,15 +27,24 @@ const ArtistInsightsAuctionResults: React.FC<Props> = ({ artist, relay, scrollTo
...
     setFilterTypeAction("auctionResult")
   }, [])
 
+  useEffect(() => {
+    setFiltersCountAction({
+      total: auctionTotalCount,
+      followedArtists: null,
+    })
+  }, [auctionTotalCount])
+
```
Comment (dimatretyak) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx:47]: We record the total number of works, and then read it in the component `ArtistInsightsFilterHeader.tsx`

File: src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx:85]: If you have the opportunity and time, please test it on real devices and evaluate the performance

Comment (olerichter00) [reply src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx:85]: Looks pretty good to me! Not sure if we use another pattern for sticky headers somewhere though...

**Pull request closed without merging**